### ISSUE TITLE
sync with native bookmarks?

### ISSUE THREAD

**Issue by developer91234** [hoarder-app/hoarder#712]
### Describe the feature you'd like

The hoarder bookmarks should be duplicated in the browser's native bookmarks, perhapas with `#tag` injected into the bookmark title. And vice versa.

### Describe the benefits this would bring to existing Hoarder users

Just easier to use native bookmarks in the browser interface, easier to move to another browser etc. Most of the time, I'd use something else entirely for saving geniunely valuable info (like obsidian or a reference manager), while for routine access to junk bookmarks and simple links Hoarders interface is too heavy-duty.

### Can the goal of this request already be achieved via other means?

Floccus does something like this if you run it with linkwarden. Maybe a plugin exists for obsidian.

### Have you searched for an existing open/closed issue?

- [X] I have searched for existing issues and none cover my fundamental request

**Comment by The-Istar** [comment]
+1 for this. This is the one thing keeping me from using. I don't always have access to my server when I am underway, this making hoarder useless in that situation.
Besides that use case there is also the worry of losing access your bookmarks whenever the server is down.
Imagine your server breaking down and not even having access anymore to all your bookmarks.
Thus resulting in having to manually keep 2 copies of important bookmarks. Both in your browser and in Hoarder. Not an ideal situation at all.

**Comment by jeffmarder** [comment]
This would be great. There's a feature request over in the floccus repository to add hoarder support. I think that would be the most ideal solution. Floccus has a pretty clean architecture where new adapters can be added and hoarder already has an API. It probably wouldn't be a ton of work to submit a PR to floccus to add this functionality.

[https://github.com/floccusaddon/floccus/issues/1745](https://github.com/floccusaddon/floccus/issues/1745)

**Comment by MohamedBassem** [comment]
@jeffmarder I had a quick look at floccus and seems like it should be fairly straightforward to implement. I'll add it to the roadmap.

**Comment by MohamedBassem** [comment]
Ok, I've submitted a PR to the floccus repo adding karakeep sync (https://github.com/floccusaddon/floccus/pull/1953), sorry it took too long.

**Comment by cirrusflyer** [comment]
Seems floccus never updated to support this?

**Comment by MadMaximusJB** [comment]
> Seems floccus never updated to support this?

From the issue thread, looks like it's coming in the next few weeks.

**Comment by MohamedBassem** [comment]
FYI, just got released yesterday in floccus: https://github.com/floccusaddon/floccus/releases/tag/v5.6.0

**Comment by cirrusflyer** [comment]
Except in the setup, there's no option for Karakeep.

**Comment by stvsu** [comment]
Has anyone found a way to automatically exclude all Floccus items from showing in the Home view in Karakeep?

I thought Rules could solve this: auto archive, add "floccus" tag. But Rules don't cascade down to the tree, its only at the "root" level

My ideal setup in Karakeep: "Floccus" list --> subfolders for Browser Profile 1, Browser Profile 2. Each of those profiles has N number of items, which can include additional folders.

Any "rules" created at the Floccus list level don't apply to the sublists, and it's impossible to manually create individual rules for the entire tree. Browser Profile 1 will have sub-lists based on how you organize bookmarks/tabs/etc.

Curious if anyone has solved this!

**Comment by MohamedBassem** [comment]
Yeah, if floccus doesn't allow you to set a predefined tag at source, it'll probably not be possible to do this in karakeep today without listing all the lists. https://github.com/karakeep-app/karakeep/pull/2620 adds support for having multiple lists in one rule, but still it means that you need to manually include all lists.

Having a rule that says: "If the bookmark is a member of this list or any of its children" (or a checkbox beside list rule to include children) sounds like a reasonable feature request to me.

**Comment by stvsu** [comment]
I do think having a "cascade" checkbox when creating a rule would be great.

I will mention on the floccus side too, they could add metadata in the API call that upserts the bookmarks (`floccus` tag; I also see `source` is available in karakeep API although I'm not sure the Rules Engine can reference that field)

> I will mention on the floccus side too

EDIT: welp it looks like you built the integration on floccus side too hehe
